src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1106,7 +1106,16 @@ public async Task<ServerNode> HandleServerNotResponsive(string url, ServerNode c
             SpawnHealthChecks(chosenNode, nodeIndex);
             _nodeSelector?.OnFailedRequest(nodeIndex);
             var (_, serverNode) = await GetPreferredNode().ConfigureAwait(false);
-            await UpdateTopologyAsync(serverNode, 0, true, debugTag: "handle-server-not-responsive").ConfigureAwait(false);
+            try
+            {
+                await UpdateTopologyAsync(serverNode, 0, true, debugTag: "handle-server-not-responsive").ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                //We don't want to stop observe for changes if server down. we will wait for one to be up
```

Comment:
I think you mix classes responsibilities. You are in  HandleServerNotResponsive method in request executor, but comment is about changes api. So I think we should handle that in changes api instead.